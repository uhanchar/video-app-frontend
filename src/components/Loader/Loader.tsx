import React, { FC } from 'react';
import { CircularProgress } from '@material-ui/core';

import 'components/Loader/Loader.scss';

interface ILoader {
  size: string | number;
}

const Loader: FC<ILoader> = ({ size }) => (
  <div className="loader-wrapper">
    <CircularProgress size={size} />
  </div>
);

export default Loader;
