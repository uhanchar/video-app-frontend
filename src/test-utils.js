import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { applyMiddleware, createStore } from 'redux';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';

import rootReducer from './reducers';

const reducerInitialState = {};

function render(
  ui,
  {
    initialState = reducerInitialState,
    store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {},
) {
  // eslint-disable-next-line react/prop-types
  function Wrapper({ children }) {
    return (
      // eslint-disable-next-line react/jsx-filename-extension
      <Provider store={store}>
        <BrowserRouter>
          {children}
        </BrowserRouter>
      </Provider>
    );
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

export * from '@testing-library/react';

export { render };
