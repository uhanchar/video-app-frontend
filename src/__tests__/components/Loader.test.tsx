import React from 'react';

import { render as renderWithRedux } from 'test-utils';
import Loader from 'components/Loader/Loader';

const size = 48;

describe('test for Loader component', () => {
  it('should render Loader component', () => {
    const { asFragment } = renderWithRedux(<Loader size={size} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render Loader of proper size according to props', () => {
    const { getByTestId } = renderWithRedux(<Loader size={size} />);

    expect(getByTestId('loader')).toHaveStyle({ width: `${ size }px`, height: `${ size }px` });
  });
});
