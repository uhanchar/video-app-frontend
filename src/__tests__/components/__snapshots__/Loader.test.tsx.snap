// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test for Loader component should render Loader component 1`] = `
<DocumentFragment>
  <div
    class="loader-wrapper"
  >
    <div
      class="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      data-testid="loader"
      role="progressbar"
      style="width: 48px; height: 48px;"
    >
      <svg
        class="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
